vision_agent_tools/models/model_registry.py

```diff
@@ -1,19 +1,50 @@
-from typing import Dict, Callable
+from typing import List, Dict, Type, Callable
+from enum import Enum
 from vision_agent_tools.shared_types import BaseMLModel
 from vision_agent_tools.models.florencev2 import Florencev2
+from vision_agent_tools.models.owlv2 import Owlv2
 
 
-MODEL_REGISTRY: Dict[str, Callable[[], BaseMLModel]] = {"florence2": Florencev2}
+class TextToObjectDetectionModel(str, Enum):
+    FLORENCEV2 = "florencev2"
+    OWLV2 = "owlv2"
 
 
-def get_model(model_name: str) -> BaseMLModel:
+MODEL_REGISTRY: Dict[str, Callable[[], BaseMLModel]] = {
+    "florencev2": Florencev2,
+    "owlv2": Owlv2,
+}
+
+TASK_MODEL_MAP: Dict[Enum, List[str]] = {
+    TextToObjectDetectionModel: [model.value for model in TextToObjectDetectionModel]
+}
+
+
+def get_model_class(model_name: str, task: Type[Enum]) -> BaseMLModel:
     """
-    Retrieve a model from the registry based on the model name
+    Retrieve a model from the registry based on the model name and task
+
+    Args:
+        model_name (str): The name of the model to retrieve
+        task (Type[Enum]): The enum representing the valid models for a specific task
+
+    Returns:
+        BaseMLModel: An instance of the requested model
+
+    Raises:
+        ValueError: If the model is not registered or is not valid for the given task
     """
 
+    if model_name not in TASK_MODEL_MAP[task]:
+        raise ValueError(
+            f"Model '{model_name}' is not valid for the task '{task.__name__}'."
+        )
+
     model_class = MODEL_REGISTRY.get(model_name)
 
     if not model_class:
-        raise ValueError(f"Model '{model_name}' is not registered")
+        raise ValueError(
+            f"Model '{model_name}' is not registered in the model registry."
+        )
 
-    return model_class()
+    return model_class
```

Comment:
I'm not sure I like this approach too much. How are we gonna handle composition of models for a single task?

Reply:
I was thinking the composition could be built in the models context then passed to the task context, but I understand your point: you need it to be called / suggested by the user and it should be done the other way. So I agree with you that it wont fit the usecase